src/model.jl

```diff
@@ -279,17 +279,20 @@ StaticModel(m::AbstractModel) = StaticModel(parent(m))
 
 # Model Utils
 
-_expandpars(x) = Flatten.reconstruct(parent, _expandkeys(parent), SELECT, IGNORE)
+_expandpars(x) = reconstructparams(parent, _expandkeys(parent))
 # Expand all Params to have the same keys, filling with `nothing`
 # This probably will allocate due to `union` returning `Vector`
 function _expandkeys(x)
     pars = params(x)
     allkeys = Tuple(union(map(keys, pars)...))
+    _expandkeys1(pars, Val{allkeys}())
+end
+Base.@assume_effects :foldable function _expandkeys1(pars, ::Val{Keys}) where Keys
```

Comment:
does `@assume_effects` actually do anything here? also does this lower bound us to Julia 1.11? I would be against that.

Reply:
maybe it was only the annotations of code blocks that was added in 1.11... I don't remember, I haven't really used this macro.

Reply:
Its from 1.10. I think `:foldable` is so that map over `Keys` will always compile away.